packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -13,6 +13,9 @@ export = {
   [consts.FRAMEWORK_ON_TIMEOUT_HANDLER_NAME]: onTimeout,
 };
 
+const BASE_SLEEP = 10_000;
+const MAX_TOTAL_SLEEP = 620_000;
```

Comment:
Rereading that, however, made me realize that the ordering on these checks might be somewhat off. If the lambda. is called while it is in `inactive` or `pending` state, the invocation will simply fail so a new invocation will need to be called instead of giving time for the current invocation to pass. Basically, I think that `invokeUserFunction` may need to have the following workflow:

```
Within a while loop `await invokeFunction` -> `await getFunctionResponse` -> 
    if `Configuration. LastUpdateStatus.failure && Configuration.State === inactive or pending` -> retry
    else if -> `Configuration. LastUpdateStatus.failure` i.e. it's status is active or failed -> get and throw error error (break)
    else -> return payload (break)
```

Reply:
The above could be a do-while loop so that you don't have to use breaks but obviously the actual implementation is up to you.

Reply:
Also, regarding this number, do we know in general how long it usually takes a function to return to active? That data might be good in determining this number.

Reply:
Unfortunately, we do not at this time. I should have lambdas that will become inactive in the next week or so (someone from lambda told me offline that this is ~30 days), so I can check once I have an inactive function.

Reply:
My original assessment was only partially right. I was under the impression that the Lamba function failed, not errored, but upon further digging, I see that it throws an exception. See `outbound.ts` for my revised suggestion here.